File: backend/src/main/java/com/khojdu/backend/entity/Complaint.java
package com.khojdu.backend.entity;

import com.khojdu.backend.entity.enums.ComplaintPriority;
import com.khojdu.backend.entity.enums.ComplaintStatus;
import com.khojdu.backend.entity.enums.ComplaintType;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "complaints")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Complaint {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "complainant_id", nullable = false)
    private User complainant;

    @ManyToOne
    @JoinColumn(name = "property_id")
    private Property property;

    @ManyToOne
    @JoinColumn(name = "landlord_id")
    private User landlord;

    @Enumerated(EnumType.STRING)
    @Column(name = "complaint_type", nullable = false)
    private ComplaintType complaintType;

    @Column(nullable = false)
    private String subject;

    @Column(nullable = false)
    private String description;

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "complaint_evidence", joinColumns = @JoinColumn(name = "complaint_id"))
    @Column(name = "evidence_url")
    private List<String> evidenceUrls;

    @Enumerated(EnumType.STRING)
    private ComplaintStatus status = ComplaintStatus.PENDING;

    @Enumerated(EnumType.STRING)
    private ComplaintPriority priority = ComplaintPriority.MEDIUM;

    @ManyToOne
    @JoinColumn(name = "assigned_to")
    private User assignedTo;

    @Column(name = "resolution_notes")
    private String resolutionNotes;

    @CreatedDate
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "resolved_at")
    private LocalDateTime resolvedAt;
}

File: frontend/types/complaint.ts
export type ComplaintType =
  | "FRAUD"
  | "OVERCHARGING"
  | "FAKE_LISTING"
  | "HARASSMENT"
  | "PROPERTY_MISMATCH"
  | "POOR_MAINTENANCE"
  | "OTHER"

export type ComplaintStatus = "PENDING" | "INVESTIGATING" | "RESOLVED" | "DISMISSED"

export type ComplaintPriority = "LOW" | "MEDIUM" | "HIGH" | "URGENT"

export interface Complaint {
  id: string
  complainantId?: string
  complainantName?: string
  propertyId?: string
  propertyTitle?: string
  landlordId?: string
  landlordName?: string
  complaintType: ComplaintType
  subject: string
  description: string
  evidenceUrls?: string[]
  status?: ComplaintStatus
  priority?: ComplaintPriority
  assignedToId?: string
  assignedToName?: string
  resolutionNotes?: string
  createdAt?: string
  resolvedAt?: string
}
